### DISCUSSION TITLE
Each field instance should have it's own handle.

### DISCUSSION THREAD

**Discussion by foxtrotcz** [craftcms/cms#13580] [Ideas] [answered]
Hello,
this is kinda similar to https://github.com/craftcms/cms/discussions/8497 but also little different.

Let say I have museum site or something.

**Current workflow and my problem:**
I want to have a description for each painting using Redactor field. So I create field which is just default Redactor, nothing special, name it "Painting Description" and handle "paintingDescription". Assign it to Section/Entry Type which is responsible for painting entries.

Then I want some About Us page witch description, also using default Redactor. I cannot use the first field, because it's handle is "paintingDescription". I have to create new one that is basically the same, just different handle, like "aboutUsRedactor".

Then I want another simple redactor elsewhere. Again I need to create new field which is basically the same as the previous one. And this keeps happening. Now I have tons of fields each used only in one instance. Not ideal and kinda messy.

Ok, I try to find solution to stop repeating the same field again and again.
I name the painting description field generally. For example name/handle is "radactor1".
It's little harder to write code now because I have to remember when coding painting template that "redactor1" is the description of painting. But on the other side I can use it for painting description, for About Us and anything else.
Only bad thing is I have to remember that "redactor1" handle means different things in different entry types.

But then I want to use multiple redactor fields on my painting page. So I create "redactor2", "readactor3" and redactor4".
Ok, now it's a mess. When coding it's very hard to remember what redactor number means what. I have to keep looking at Craft Admin entry page to remind myself what should I type.

**Proposed solution:**
You could create Field Type but without handle. Handle of each instance of field would be defined inside Entry Type.

So I would create regular default Redactor field type. Then I would assign it to painting page entry type. Inside this context it would be for description of painting so I would define handle "paintingDescription".
Then I would use the same Field Type for About Us page. And inside About Us entry type I would define handle of this field as "aboutUsRedactor".
And if I wanted to use multiple instances of this field in my painting page I would just assign each instance different handle according to it's purpose and context.

I think this ability to set handles specifically for each instance of field would be good.

I hope I described it somewhat understandably.

Thanks.

**Comment by brandonkelly** [comment, marked as answer]
This is exactly how we’ve implemented field instances (#8497) for Craft 5 ;)

**Reply by foxtrotcz** [reply]
Thanks, I cant wait :)
